src/dhcp6relay/src/configInterface.cpp

```diff
@@ -0,0 +1,137 @@
+    }
+    catch (const std::bad_alloc &e) {
+        syslog(LOG_ERR, "Failed allocate memory. Exception details: %s", e.what());
+    }
+}
+
+//
+// ---> deinitialize();
+//
+// deinitialize DB interface and join SWSS listening thread
+//
+void deinitialize_swss()
+{
+    stopSwssNotificationPoll();
+    mSwssThreadPtr->interrupt();
+}
+
+/**
+ * @code                void handleSwssNotification(arg_config *context)
+ * 
+ * @brief               main thread for handling SWSS notification
+ *
+ * @param context       argument config that contains strings of server and option
+ *
+ * @return              none
+ */
+void handleSwssNotification(arg_config *context)
+{
+    std::shared_ptr<swss::DBConnector> configDbPtr = std::make_shared<swss::DBConnector> ("CONFIG_DB", 0);
+    swss::SubscriberStateTable ipHelpersTable(configDbPtr.get(), "DHCP");
+
+    swss::Select swssSelect;
+    swssSelect.addSelectable(&ipHelpersTable);
+    while (mPollSwssNotifcation) {
+        swss::Selectable *selectable;
+        int ret = swssSelect.select(&selectable, DEFAULT_TIMEOUT_MSEC);
+        if (ret == swss::Select::ERROR) {
+            syslog(LOG_WARNING, "Select: returned ERROR");
+            continue;
+        } else if (ret == swss::Select::TIMEOUT) {
+            continue;
+        } 
+        if (selectable == static_cast<swss::Selectable *> (&ipHelpersTable)) {
+            handleRelayNotification(ipHelpersTable, context);
+        }
+    } 
+}
+
+/**
+ * @code                    void handleRelayNotification(swss::SubscriberStateTable &ipHelpersTable, arg_config context)
+ * 
+ * @brief                   handles DHCPv6 relay configuration change notification
+ *
+ * @param ipHelpersTable    DHCP table
+ * @param context           argument config that contains strings of server and option
+ *
+ * @return                  none
+ */
+void handleRelayNotification(swss::SubscriberStateTable &ipHelpersTable, arg_config *context)
+{
+    std::deque<swss::KeyOpFieldsValuesTuple> entries;
+
+    ipHelpersTable.pops(entries);
+    processRelayNotification(entries, context);
+}
+
+/**
+ * @code                    void processRelayNotification(std::deque<swss::KeyOpFieldsValuesTuple> &entries, arg_config context)
+ * 
+ * @brief                   process DHCPv6 relay servers and options configuration change notification
+ *
+ * @param entries           queue of std::tuple<std::string, std::string, std::vector<FieldValueTuple>> entries in DHCP table
+ * @param context           argument config that contains strings of server and option
+ *
+ * @return                  none
+ */
+void processRelayNotification(std::deque<swss::KeyOpFieldsValuesTuple> &entries, arg_config *context)
+{
+    std::vector<std::string> servers;
+
+    for (auto &entry: entries) {
+        std::string vlan = kfvKey(entry);
+        std::vector<swss::FieldValueTuple> fieldValues = kfvFieldsValues(entry);
+
+        for (auto &fieldValue: fieldValues) {
```

Comment:
we need to check the operation here if it is ADD/DEL

Reply:
The dhcpv6 server address are type string value. e.g.: "fc02:2000::1,fc02:2000::2"
HDEL would delete the entire string of addresses. HSET updates the list of server addresses.
When config_db updates the list of server addresses, all addresses are properly fetched and processed regardless new addresses are added/deleted. Therefore, I don't think there is a need to check for add/del

Reply:
You can get the operation using `std::string operation = kfvOp(entry);`. This way we can tell the type of the record updates. I think they are `SET/DEL` operation.